src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public class JsonContent : HttpContent
+    {
+        internal const string JsonMediaType = "application/json";
+
+        private readonly byte[] _content;
+        private readonly int _offset;
+        private readonly int _count;
+
+        private static MediaTypeHeaderValue CreateMediaType(string mediaTypeAsString)
+        {
+            MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse(mediaTypeAsString);
+
+            // If the instantiated mediaType does not specify its CharSet, set UTF-8 by default.
+            if (mediaType.CharSet == null)
+            {
+                mediaType.CharSet = Encoding.UTF8.WebName;
+            }
+
+            return mediaType;
+        }
+
+        public Type ObjectType { get; }
```

Comment:
When `Create<T>` is callled, this is `typeof(T)`.
When the constructor is called, this is the specified `type` argument.
As per the Formatting API, this is always the declared type. Should we do the same here? 

Note that if we do the same, this will always be `typeof(object)` when we call the constructor.